extra/xmlchecker.py

```diff
@@ -0,0 +1,43 @@
+import xmlschema
+import sys
+import xml.etree.ElementTree as ET
+import subprocess
+from pathlib import Path
+import os
+
+# change working directory in case script was not called from helios root directory
+survey_file = Path(sys.argv[1]).resolve()
+xsd_dir = Path(__file__).parent
+survey_file = str(survey_file.relative_to(xsd_dir.parent.resolve()))
+if Path.cwd() != xsd_dir.parent:
+    os.chdir(xsd_dir.parent)
+
+# check where helios.exe is located
+if any("helios.exe" in str(f) for f in Path.cwd().glob('*') if f.is_file()):
+    helios_dir = Path("")
+elif any("helios.exe" in str(f) for f in Path.cwd().glob('run/*') if f.is_file()):
+    helios_dir = Path("run")
+
+survey_schema = xmlschema.XMLSchema('extra\survey.xsd')
+scene_schema = xmlschema.XMLSchema('extra\scene.xsd')
+scanner_schema = xmlschema.XMLSchema('extra\scanner.xsd')
+platform_schema = xmlschema.XMLSchema('extra\platform.xsd')
+
+# get paths of any referenced XML files; assuming they are relative to helios root dir
+try:
+    scene_file = ET.parse(survey_file).find('survey').attrib['scene'].split('#')[0]
+    scanner_file = ET.parse(survey_file).find('survey').attrib['scanner'].split('#')[0]
+    platform_file = ET.parse(survey_file).find('survey').attrib['platform'].split('#')[0]
+except KeyError as e:
+    print("ERROR: Missing 'platform', 'scanner' or 'scene' key in <survey> tag.\n"
+          "Please check your survey file.\n")
+    raise e
+
+# validate XML files
+xmlschema.validate(survey_file, survey_schema)
+xmlschema.validate(scene_file, scene_schema)
+xmlschema.validate(scanner_file, scanner_schema)
+xmlschema.validate(platform_file, platform_schema)
+
+# call HELIOS++
+subprocess.run([helios_dir / "helios.exe", survey_file, *sys.argv[2:]])
```

Comment:
could also check for operating system:
```
HELIOS_EXE = str(Path('build') / 'helios')
if sys.platform == "win32":
    HELIOS_EXE += ".exe"
```

Reply:
We could also say the users specify the path to the helios executable. Because we do not know if their executable is in the helios root directory or in a `run` folder or in another folder (like `build`)

Reply:
We could check if it is in the standard HELIOS dir and if not, check if it is in PATH (and could be somewhere else but will be found by the system). I would say those are the two options to go for and that could be supported.